### ISSUE TITLE
Cannot generate the same validation images

### ISSUE THREAD

**Issue by fiona-lxd** [bghira/SimpleTuner#207]
There are some validation images generated during training with the helpers/prompts.py. However, when I tried to reproduce the same results during inference, it fails to generate same images. (The text encoder is frozen during training.) Can you help me to figure it out?

My inference codes are like

pipe = StableDiffusionXLPipeline.from_pretrained(
            "stable-diffusion-xl-base-1.0", torch_dtype=torch.float16, variant="fp16", use_safetensors=True
        )
pipe.unet = UNet2DConditionModel.from_pretrained(os.path.join(model_path, 'unet'))
pipe.vae = AutoencoderKL.from_pretrained(
        "sdxl-vae-fp16-fix",
        subfolder=None,
        revision=False,
        force_upcast=False,
    )
pipe.scheduler = DDIMScheduler.from_pretrained(
                        "stable-diffusion-xl-base-1.0",
                        subfolder="scheduler",
                        prediction_type="epsilon",
                        timestep_spacing="trailing",
                        rescale_betas_zero_snr=False
                    )
weight_dtype = torch.bfloat16
pipe.unet.to(dtype=weight_dtype)
pipe.text_encoder.to(dtype=weight_dtype)
pipe.text_encoder_2.to(dtype=weight_dtype)
pipe.vae.to(dtype=pipe.unet.dtype)
pipe = DiffusionPipeline.from_pretrained(model_path, torch_dtype=torch.float16)
pipe.to("cuda")
image = pipe(prompt=prompt, negative_prompt=negative_prompt, num_inference_steps=30, generator = torch.Generator(device="cuda").manual_seed(420420420), guidance_scale=7.5, guidance_rescale=0.0, height=1024, width=1024).images[0]

prompt : "autumn portrait of a russian young woman inside cafe, knitted fox-hat, fog close-up, storm clouds, wearing red sunglasses, red shirt, (artstation, deviantart)"
negative_prompt: "blurry, cropped, ugly"

validation during training:
All the settings are the same with the training scripts including the seed. But the generated images are slightly different as follows:
![step_100_val_img_3](https://github.com/bghira/SimpleTuner/assets/58323411/55de4af4-1044-4bbc-be21-f6face8e9d70)

validation with the above inference code:
![tmp - 2023-10-09T172900 618](https://github.com/bghira/SimpleTuner/assets/58323411/06d28e12-5b99-4dd5-a6ac-ea984a09ea0b)

**Comment by bghira** [comment]
you could try `device('cpu')` for the Generator, then. maybe `timestep_spacing='leading'`, since that's the default SDXL 1.0 config.

**Comment by fiona-lxd** [comment]
I tried but still failed. So weird.

**Comment by bghira** [comment]
mm, we do use torch.autocast - but i'm not terribly sure where the difference would lie. the text embed vector is saved at startup, and then reused on each validation. perhaps the seed of this embed vector matters too, and we are not using the validation seed for that - the random states shift as the embeds are made.
